networkx/algorithms/centrality/tests/test_trophic.py

```diff
@@ -300,3 +300,9 @@ def test_trophic_incoherence_parameter_cannibalism():
     q = nx.trophic_incoherence_parameter(G, cannibalism=True)
     # Ignore the -link
     assert q == pytest.approx(np.std([1, 1.5, 0.5, 0.75, 1.25]), abs=1e-7)
+
+
+def test_no_basal_node():
+    G = nx.DiGraph([(1, 2), (2, 3), (3, 1)])  # No basal node, should raise an error
+    with pytest.raises(nx.NetworkXError):
+        nx.trophic_levels(G)
```

Comment:
That said, +1 for adding a test to probe the reachability check!

Reply:
I would have thought the bfs would fail because the only basal node is `4`. So the BFS starts (and ends) there. How do the other nodes become part of the BFS? I think I'm missing something... We're trying to test whether all nodes are reachable from a basal node, right?

Reply:
Upon closer review, you're completely right :) I missed the subtle difference in the edges (`[(0, 1), (1, 2), (2, 3)]` vs. `[(0, 1), (1, 2), (2, 3)]`) between the two examples. Thanks for bearing with me :upside_down_face: